### PR TITLE
Issue/318 dims index

### LINKED ISSUE #318: Coordinates should be changed to `range(len(original_coords))` just before creating a batch
## Detailed Description
Different DataSources do this slightly differently.  We should harmonise.  My proposal is that `make_dim_index` should be called as a final step in `get_batch()`.

For example, TopographicDataSource changes the coords in `get_example`.

And a few other little tweaks:

* Maybe remove `xr_utils.convert_data_array_to_dataset()` and instead call `to_dataset()` and `make_dim_index()` separately in calling code, so the calling code makes it clear when the index is being changed?
* Maybe rename `xr_utils.make_dim_index()`?
* Document where coords get changed.

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

Tidy up where coords get changed to indexes

Fixes #318, #372

## How Has This Been Tested?

Unittest tests have not been changed, (but they definitely caught a few development bugs)

- [ ] No
- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: nowcasting_dataset/data_sources/nwp/nwp_data_source.py [hunk 1]
```diff
@@ -108,10 +113,10 @@ def _get_time_slice(self, t0_dt: pd.Timestamp) -> xr.DataArray:
         selected["target_time"] = init_time + selected.step
         return selected
 
-    def _post_process_example(
-        self, selected_data: xr.DataArray, t0_dt: pd.Timestamp
-    ) -> xr.DataArray:
+    def _post_process_example(self, selected_data: xr.Dataset, t0_dt: pd.Timestamp) -> xr.Dataset:
         """Resamples to 5 minutely."""
+
+        print(selected_data)
```
Comment (JackKelly) [inline nowcasting_dataset/data_sources/nwp/nwp_data_source.py:119]: Remove this print() statement?

File: nowcasting_dataset/dataset/xr_utils.py [hunk 2]
```diff
@@ -29,18 +20,16 @@ def join_list_dataset_to_batch_dataset(datasets: list[xr.Dataset]) -> xr.Dataset
...
-    data = xr.Dataset({"data": data_xarray})
-    return make_dim_index(dataset=data)
+def convert_coordinates_to_indexes_for_list_datasets(
+    examples: List[xr.Dataset],
+) -> List[xr.Dataset]:
+    """Set the coords to be indices before joining into a batch"""
+    return [convert_coordinates_to_indexes(example) for example in examples]
 
 
 # TODO: Issue #318: Maybe rename this function... maybe to coord_to_range()?
```
Comment (JackKelly) [inline nowcasting_dataset/dataset/xr_utils.py:30]: Also delete this comment?

**Review by JackKelly — APPROVED** [review]
LGTM! Thanks!

**Review by jacobbieker — APPROVED** [review]
LGTM!

**Pull request merged**